src/Groups/types.jl

```diff
@@ -198,6 +214,68 @@ struct AutomorphismGroup{T} <: GAPGroup
   end
 end
 
+################################################################################
+#
+#  Composite Groups
+#
+################################################################################
+
+
+"""
+    DirectProductOfGroups
+Either direct product of two or more groups of any type, or subgroup of a direct product of groups.
+"""
+struct DirectProductOfGroups <: GAPGroup
```

Comment:
Yes, the fact that groups of type `DirectProductGroup` are not always direct products of groups, but they could be proper subgroups. If there are only two factors, then every subgroup of the direct product is a `SubdirectProductGroup`, as defined in https://www.gap-system.org/Manuals/doc/ref/chap49.html#X815AFC537B215D7B, so using subdirect products covers all the cases. I don't know if this extends to products of more than two factors.

Reply:
Side remark: with the definition of subdirect product I use, every subgroup U of a direct product GxH is a subdirect product; but not necessarily a subdirect product **of GxH**, but rather it will be a subdirect product of AxB, where A is the projection of U to G, and B is the projection of U to H. This is another reason why I am somewhat wary of using `SubdirectProductGroup` as a name. Anyway, there's probably no good solution here...

... and it remains to see how useful these "external direct products" are anyway... in all applications I worked on in the past, it was almost always better to avoid these, and e.g. represent the direct product of permutation groups as a permutation group. So perhaps we shouldn't spend *too* much time on tweaking these...